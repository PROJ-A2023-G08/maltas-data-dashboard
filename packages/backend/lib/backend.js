'use strict';

module.exports = backend;

function backend() {
  return 'Hello from backend';
}
