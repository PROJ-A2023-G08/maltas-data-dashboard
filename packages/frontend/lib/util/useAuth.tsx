import { useRouter } from 'next/router';
import { useEffect, useState } from 'react';

const useAuth = () => {
  const router = useRouter();
  const [accessToken, setAccessToken] = useState<string | undefined | null>();
  const [refreshToken, setRefreshToken] = useState<string | undefined>();
  const [isLoggedIn, setIsLoggedIn] = useState(false);

  useEffect(() => {
    // If accessToken and refreshToken are not provided as props, try to get them from localStorage
    const storedAccessToken = localStorage.getItem('accessToken');
    if (storedAccessToken) {
        setAccessToken(storedAccessToken);
        setIsLoggedIn(true);
    }
    const storedRefreshToken = localStorage.getItem('refreshToken');
    if (storedRefreshToken) {
        setRefreshToken(storedRefreshToken);
    }

    // Check if the page requires authentication
    const exemptPaths = ['/login', '/register','/about','/contact','/home'];
    const requiresAuth = !exemptPaths.includes(router.pathname);

    if (requiresAuth && !storedAccessToken) {
      // Redirect to home page if not authenticated
      router.push('/home');
    }
  }, [accessToken, refreshToken, router]);

  const logout = () => {
    // Clear tokens from storage
    localStorage.removeItem('accessToken');
    localStorage.removeItem('refreshToken');
    setIsLoggedIn(false);
    
    // Redirect to the home page after logout
  
    router.push('/home');
  };

  return { isLoggedIn, accessToken, refreshToken, logout };
};

export default useAuth;
