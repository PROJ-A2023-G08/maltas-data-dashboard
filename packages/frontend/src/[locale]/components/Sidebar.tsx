import React from "react";
import List from "@mui/material/List";
import ListItem from "@mui/material/ListItem";
import ListItemButton from "@mui/material/ListItemButton";
import ListItemIcon from "@mui/material/ListItemIcon";
import ListItemText from "@mui/material/ListItemText";
import HomeIcon from "@mui/icons-material/Home";
import GridViewIcon from "@mui/icons-material/GridView";
import SettingsIcon from "@mui/icons-material/Settings";
import HelpIcon from "@mui/icons-material/Help";
import LogoutIcon from "@mui/icons-material/Logout";

interface SidebarProps {
  setActiveComponent: (componentName: string) => void;
}

const Sidebar: React.FC<SidebarProps> = ({ setActiveComponent }) => {
  const handleSidebarItemClick = (componentName: string) => {
    setActiveComponent(componentName);
  };

  const menuItems = [
    {
      name: "Home",
      icon: <HomeIcon />,
    },
    {
      name: "Dashboard",
      icon: <GridViewIcon />,
    },
    {
      name: "Settings",
      icon: <SettingsIcon />,
    },
    {
      name: "Help",
      icon: <HelpIcon />,
    },
  ];

  return (
    <div className="bg-gray-800 text-white w-60 min-h-screen">
      <div className="flex flex-col items-start justify-between h-full">
        <div className="flex flex-col items-start justify-start w-full">
          <div className="flex flex-col items-center justify-center w-full h-40">
            <img
              className="w-8 h-8 rounded-full"
              src="https://avatars.githubusercontent.com/u/10491406?v=4"
              alt="avatar"
            />
            <h3 className="text-lg font-bold">LOGO HERE</h3>
          </div>
          <div className="flex flex-col items-start justify-start w-full">
            <List component="nav" aria-label="main mailbox folders">
              {menuItems.map((item) => (
                <ListItem key={item.name}>
                  <ListItemButton
                    onClick={() => handleSidebarItemClick(item.name)}
                  >
                    <ListItemIcon>{item.icon}</ListItemIcon>
                    <ListItemText primary={item.name} />
                  </ListItemButton>
                </ListItem>
              ))}
            </List>
          </div>
        </div>
        <div className="flex flex-row items-center justify-start w-full h-12 pl-5 cursor-pointer hover:bg-gray-700">
          <div className="mr-4">
            <LogoutIcon />
          </div>
          <div>Log out here?</div>
        </div>
      </div>
    </div>
  );
};

export default Sidebar;
