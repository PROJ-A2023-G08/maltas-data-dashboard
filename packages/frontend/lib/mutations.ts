import * as api from "./api/api";
import { useMutation, useQueryClient } from 'react-query';
import { qkGetUserProfile, qkAllUsers } from "./api/queryKeys";
import { UpdatePasswordQueryParams, UpdateUserQueryParams, UploadProfileImageQueryParams, AdminDeleteUserQueryParams, AdminUpdateUserRoleQueryParams  } from "./types/queryTypes";
import useAuth from "./util/useAuth";
import { User } from "./types";

export const useMUpdatePasswordMutation = () => {
  const { accessToken, fastAccessToken}= useAuth()
  const queryClient = useQueryClient();
  return useMutation(
    (data: UpdatePasswordQueryParams) => api.updatePassword(fastAccessToken || accessToken!, data),
    {
      onSuccess: () => {
        queryClient.invalidateQueries(qkGetUserProfile);
      },
    }
  );
};

export const useMUploadProfileImageMutation = () => {
  const queryClient = useQueryClient();
  const { accessToken}= useAuth()
  return useMutation(
    (data: UploadProfileImageQueryParams) => api.uploadProfileImage(accessToken!, data),
    {
      onSuccess: () => {
        queryClient.invalidateQueries(qkGetUserProfile);
      },
    }
  );
};

export const useMUpdateUserProfileMutation = () => {
  const queryClient = useQueryClient();
  const { accessToken, fastAccessToken}= useAuth()
  return useMutation(
    (data: UpdateUserQueryParams) => api.updateUserProfile(fastAccessToken || accessToken!, data),
    {
      onSuccess: (newUserData: User) => {
        queryClient.invalidateQueries(qkGetUserProfile);
      },
    }
  );
};

export const useMUpdateUserRoleMutation = () => {
  const queryClient = useQueryClient();
  const { accessToken, fastAccessToken}= useAuth()
  return useMutation(
    (data: AdminUpdateUserRoleQueryParams) => api.updateUserRole(fastAccessToken || accessToken!, data),
    {
      onSuccess: () => {
        queryClient.invalidateQueries(qkGetUserProfile);
      },
    }
  );
};

export const useMDeleteUserMutation = () => {
  const queryClient = useQueryClient();
  const { accessToken, fastAccessToken}= useAuth()
  return useMutation(
    (data: AdminDeleteUserQueryParams) => api.deleteUser(fastAccessToken ||accessToken!, data),
    {
      onSuccess: () => {
        queryClient.invalidateQueries(qkAllUsers);
      },
    }
  );
};


