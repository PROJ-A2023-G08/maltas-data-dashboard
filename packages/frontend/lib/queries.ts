import { useQuery, UseQueryOptions, UseQueryResult } from "react-query";
import * as api from "./api/api";
import {
  User,
  AdminGetSingleUserResult,
  AdminGetAllUsersResult,
} from "./types";
import useAuth from "./util/useAuth";
import {
  qkGetUserProfile,
  qkAllUsers,
  qkGetSingleUserByAdmin,
} from "./api/queryKeys";
import { AdminGetSingleUserQueryParams, GetMeasurement } from "./types/queryTypes";

export const useUserProfile = (options?: UseQueryOptions<User>) => {
  const { accessToken, fastAccessToken } = useAuth();
  return useQuery<User>(
    [qkGetUserProfile, accessToken],
    () => api.getUserProfile(fastAccessToken || accessToken!),
    options,
  );
};

export const useAdminGetAllUsers = (
  options?: UseQueryOptions<AdminGetAllUsersResult>,
) => {
  const { accessToken, fastAccessToken } = useAuth();
  return useQuery<AdminGetAllUsersResult>(
    [qkAllUsers, accessToken],
    () => api.getAllUsers(fastAccessToken || accessToken!),
    options,
  );
};

export const useAdminGetSingleUser = (
  data: AdminGetSingleUserQueryParams,
  options?: UseQueryOptions<AdminGetSingleUserResult>,
) => {
  const { accessToken, fastAccessToken } = useAuth();
  return useQuery<AdminGetSingleUserResult>(
    [qkGetSingleUserByAdmin, accessToken],
    () => api.getSingleUser(fastAccessToken || accessToken!, data),
    options,
  );
};

export const useGetCSVData = (
  options?: UseQueryOptions<GetMeasurement>,
) => {
  const { accessToken, fastAccessToken } = useAuth();
  return useQuery<GetMeasurement>(
    ['csv', accessToken],
    () => api.getCSVData(fastAccessToken || accessToken!),
    options,
  );
};
