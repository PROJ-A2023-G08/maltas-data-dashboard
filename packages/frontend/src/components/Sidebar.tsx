import React, { useState } from "react";
import List from "@mui/material/List";
import ListItem from "@mui/material/ListItem";
import ListItemButton from "@mui/material/ListItemButton";
import ListItemIcon from "@mui/material/ListItemIcon";
import ListItemText from "@mui/material/ListItemText";
import HomeIcon from "@mui/icons-material/Home";
import GridViewIcon from "@mui/icons-material/GridView";
import SettingsIcon from "@mui/icons-material/Settings";
import HelpIcon from "@mui/icons-material/Help";
import LogoutIcon from "@mui/icons-material/Logout";
import { useTranslation } from "next-i18next";
import useAuth from "../../lib/util/useAuth";
import Image from 'next/image'
import { ButtonBase } from "@mui/material";
import { useUserProfile } from "../../lib/queries";
import { Typography, Avatar } from "@mui/material";

interface SidebarProps {
  setActiveComponent: (componentName: string) => void;
}

const Sidebar: React.FC<SidebarProps> = ({ setActiveComponent }) => {
  const { t } = useTranslation("common");
  const { logout} = useAuth();
  const userDataApi = useUserProfile();
  const userData = userDataApi?.data 

  const [selectedItem, setSelectedItem] = useState<string>('');

  const handleSidebarItemClick = (componentName: string) => {
    setActiveComponent(componentName);
    setSelectedItem(componentName); 
  };

  const menuItems = [
    {
      name: "Home",
      icon: <HomeIcon />,
    },
    {
      name: "Dashboard",
      icon: <GridViewIcon />,
    },
    {
      name: "Settings",
      icon: <SettingsIcon />,
    },
    {
      name: "Help",
      icon: <HelpIcon />,
    },
  ];

  return (
    <div className="bg-darkblue text-white w-60 min-h-screen pl-2 pr-2">
      <div className="flex flex-col items-start justify-between h-full">
        <div className="flex flex-col items-start justify-start w-full">
          <div className="flex flex-col items-center justify-center w-full h-36">
            <div className="pt-20">
              <Image
                src="/Malta_logo.svg"
                alt="Maltas Logo"
                width={160}
                height={160}
              />
            </div>
          </div>

          <div className="flex flex-col items-start justify-start w-full pt-12">
            <Typography color={"primary"} variant="h5" sx={{
              width: "100%",
              textAlign: "center",
              marginTop: "15px",
              fontWeight: 700,
            }}>
              Maltas technology Oy
            </Typography>
            {userData && (
              <List sx={{
                paddingTop: "20px"
              }}>
                <ListItem>
                  <ListItemIcon>
                    <Avatar alt={"user image"} src={userData.imageUrl || ""} />
                  </ListItemIcon>
                  <ListItemText
                    sx={{
                      fontWeight: 700,
                    }}
                    primary={`${userData.firstName || ""}  || ${
                      userData.lastName || ""
                    }`}
                    color="#fff"
                    secondary={
                      <Typography variant="body2" color={"primary.main"}>
                        {userData.email || ""}
                      </Typography>
                    }
                  />
                </ListItem>
              </List>
            )}
            <List
              className="w-full"
              component="nav"
              aria-label="main mailbox folders"
            >
              {menuItems.map((item) => (
                <ListItem
                  sx={{
                    width: "100% !important",
                    cursor: "pointer",
                  }}
                  onClick={() => handleSidebarItemClick(item.name)}
                  key={item.name}
                  className={`${
                    selectedItem === item.name
                      ? "bg-selection text-black rounded-xl mb-2"
                      : "hover:bg-white hover:text-black rounded-xl mb-2"
                  }`}
                  style={{ height: "70px" }}
                >
                  <ButtonBase
                    className="flex items-center"
                    disableTouchRipple
                    disableRipple
                  >
                    <div className="-ml-2 pt-1">
                      <ListItemIcon>{item.icon}</ListItemIcon>
                    </div>
                    <p className="font-bold text-xl -ml-6">
                      {t("sidebar." + item.name)}
                    </p>
                  </ButtonBase>
                </ListItem>
              ))}
            </List>
          </div>
        </div>
        <div
          onClick={() => {
            logout();
          }}
          className="flex flex-row items-center justify-start w-52 h-12 pl-5 cursor-pointer hover:bg-white hover:text-black hover:rounded-md"
        >
          <div className="mr-4">
            <LogoutIcon />
          </div>
          <p className="font-bold">{t("sidebar.LogOut")}</p>
        </div>
      </div>
    </div>
  );
};

export default Sidebar;