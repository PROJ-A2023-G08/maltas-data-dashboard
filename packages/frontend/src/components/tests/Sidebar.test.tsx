import React from "react";
import Sidebar from "../Sidebar";
import { render, screen } from "@testing-library/react";
import useAuth from "../../../lib/util/useAuth";
import "@testing-library/jest-dom";

jest.mock("../../../lib/util/useAuth");
jest.mock("../../../lib/queries");

// Mock the useTranslation hook from i18next to return a translation function
jest.mock("next-i18next", () => ({
  useTranslation: jest.fn().mockReturnValue({
    // @ts-ignore
    t: (str) => str,
  }),
}));

// Declare userData and mock setup outside the test
const userData = {
  firstName: "Tester",
  lastName: "Testerson",
  email: "tester.test@test.com",
  imageUrl: "https://example.com/image.jpg",
};

beforeEach(() => {
  // Set up the mock implementation for useAuth
  (useAuth as jest.MockedFunction<typeof useAuth>).mockReturnValue({
    isLoggedIn: true,
    accessToken: "your-access-token",
    refreshToken: "your-refresh-token",
    logout: jest.fn(),
    fastAccessToken: "your-fast-access-token",
  });

  // Set up the mock implementation for useUserProfile
  require("../../../lib/queries").useUserProfile.mockReturnValueOnce({
    data: userData,
  });
});

describe("Sidebar", () => {
  it("logo renders", () => {
    render(<Sidebar setActiveComponent={() => {}} />);
    expect(screen.getByTestId("maltas-logo")).toBeInTheDocument();
  });
  it("User names render", () => {
    render(<Sidebar setActiveComponent={() => {}} />);
    expect(screen.getByTestId("user-names")).toBeInTheDocument();
  });
  it("User email render", () => {
    render(<Sidebar setActiveComponent={() => {}} />);
    screen.debug();
    expect(screen.getByText("tester.test@test.com")).toBeInTheDocument();
  });
});
