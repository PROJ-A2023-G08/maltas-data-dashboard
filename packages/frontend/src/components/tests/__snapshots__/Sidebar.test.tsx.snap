// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly 1`] = `
<div
  className="bg-darkblue text-white w-60 min-h-screen pl-2"
>
  <div
    className="flex flex-col items-start justify-between h-full"
  >
    <div
      className="flex flex-col items-start justify-start w-full"
    >
      <div
        className="flex flex-col items-center justify-center w-full h-40"
      >
        <div
          className="pt-20"
        >
          <img
            alt="Maltas Logo"
            data-nimg="1"
            decoding="async"
            height={160}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/Malta_logo.svg"
            style={
              {
                "color": "transparent",
              }
            }
            width={160}
          />
        </div>
      </div>
      <div
        className="flex flex-col items-start justify-start w-full pt-36"
      >
        <nav
          aria-label="main mailbox folders"
          className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        >
          <li
            className="MuiListItem-root MuiListItem-gutters MuiListItem-padding hover:bg-white hover:text-black rounded-xl mb-2 css-1p823my-MuiListItem-root"
            disabled={false}
            style={
              {
                "height": "70px",
                "width": "200px",
              }
            }
          >
            <button
              className="MuiButtonBase-root flex items-center css-10d1a0h-MuiButtonBase-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <div
                className="-ml-2 pt-1"
              >
                <div
                  className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="HomeIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                    />
                  </svg>
                </div>
              </div>
              <p
                className="font-bold text-xl -ml-6"
              >
                sidebar.Home
              </p>
            </button>
          </li>
          <li
            className="MuiListItem-root MuiListItem-gutters MuiListItem-padding hover:bg-white hover:text-black rounded-xl mb-2 css-1p823my-MuiListItem-root"
            disabled={false}
            style={
              {
                "height": "70px",
                "width": "200px",
              }
            }
          >
            <button
              className="MuiButtonBase-root flex items-center css-10d1a0h-MuiButtonBase-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <div
                className="-ml-2 pt-1"
              >
                <div
                  className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="GridViewIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 3v8h8V3H3zm6 6H5V5h4v4zm-6 4v8h8v-8H3zm6 6H5v-4h4v4zm4-16v8h8V3h-8zm6 6h-4V5h4v4zm-6 4v8h8v-8h-8zm6 6h-4v-4h4v4z"
                      fillRule="evenodd"
                    />
                  </svg>
                </div>
              </div>
              <p
                className="font-bold text-xl -ml-6"
              >
                sidebar.Dashboard
              </p>
            </button>
          </li>
          <li
            className="MuiListItem-root MuiListItem-gutters MuiListItem-padding hover:bg-white hover:text-black rounded-xl mb-2 css-1p823my-MuiListItem-root"
            disabled={false}
            style={
              {
                "height": "70px",
                "width": "200px",
              }
            }
          >
            <button
              className="MuiButtonBase-root flex items-center css-10d1a0h-MuiButtonBase-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <div
                className="-ml-2 pt-1"
              >
                <div
                  className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="SettingsIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z"
                    />
                  </svg>
                </div>
              </div>
              <p
                className="font-bold text-xl -ml-6"
              >
                sidebar.Settings
              </p>
            </button>
          </li>
          <li
            className="MuiListItem-root MuiListItem-gutters MuiListItem-padding hover:bg-white hover:text-black rounded-xl mb-2 css-1p823my-MuiListItem-root"
            disabled={false}
            style={
              {
                "height": "70px",
                "width": "200px",
              }
            }
          >
            <button
              className="MuiButtonBase-root flex items-center css-10d1a0h-MuiButtonBase-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <div
                className="-ml-2 pt-1"
              >
                <div
                  className="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden={true}
                    className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="HelpIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 17h-2v-2h2v2zm2.07-7.75-.9.92C13.45 12.9 13 13.5 13 15h-2v-.5c0-1.1.45-2.1 1.17-2.83l1.24-1.26c.37-.36.59-.86.59-1.41 0-1.1-.9-2-2-2s-2 .9-2 2H8c0-2.21 1.79-4 4-4s4 1.79 4 4c0 .88-.36 1.68-.93 2.25z"
                    />
                  </svg>
                </div>
              </div>
              <p
                className="font-bold text-xl -ml-6"
              >
                sidebar.Help
              </p>
            </button>
          </li>
        </nav>
      </div>
    </div>
    <div
      className="flex flex-row items-center justify-start w-52 h-12 pl-5 cursor-pointer hover:bg-white hover:text-black hover:rounded-md"
    >
      <div
        className="mr-4"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="LogoutIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m17 7-1.41 1.41L18.17 11H8v2h10.17l-2.58 2.58L17 17l5-5zM4 5h8V3H4c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h8v-2H4V5z"
          />
        </svg>
      </div>
      <p
        className="font-bold"
      >
        sidebar.LogOut
      </p>
    </div>
  </div>
</div>
`;
